apps/web/modules/api/management/lib/helper.ts

```diff
@@ -0,0 +1,27 @@
+import { getResponseSurveyId, getSurveyEnvironmentId } from "@/modules/api/management/lib/services";
+import { ApiErrorResponse } from "@/modules/api/types/api-error";
+import { Result, ok } from "@formbricks/types/error-handlers";
+
+export const getEnvironmentIdFromSurveyId = async (
+  surveyId: string
+): Promise<Result<string, ApiErrorResponse>> => {
+  const surveyResult = await getSurveyEnvironmentId(surveyId);
+
+  if (!surveyResult.ok) {
+    return surveyResult;
+  }
+
+  return ok(surveyResult.data.environmentId);
+};
+
+export const getEnvironmentIdFromResponseId = async (
+  responseId: string
+): Promise<Result<string, ApiErrorResponse>> => {
+  const responseResult = await getResponseSurveyId(responseId);
+
+  if (!responseResult.ok) {
+    return responseResult;
+  }
+
+  return await getEnvironmentIdFromSurveyId(responseResult.data.surveyId);
```

Comment:
why making two database calls getting the surveyId first and then making another database call to get the environmentId. We should do this in one database call.

Reply:
Ahh, okay, we also have this similar pattern in the action client. The idea was to write small reusable functions and not make complex nested database queries.